tests/stdlib/thttpclient.nim

```diff
@@ -112,6 +114,40 @@ proc syncTest() =
   except:
     doAssert false, "TimeoutError should have been raised."
 
-syncTest()
+proc makeIPv6HttpServer(hostname: string, port: Port): AsyncFD =
+  let fd = newNativeSocket(AF_INET6)
+  setSockOptInt(fd, SOL_SOCKET, SO_REUSEADDR, 1)
+  var aiList = getAddrInfo(hostname, port, AF_INET6)
```

Comment:
I don't suppose you would be willing to implement a ``dial``-like procedure for server sockets? (Golang calls it ``listen`` IIRC) The reason I ask is because I would prefer if the tests didn't use so many low-level calls.

Of course, feel free to refuse :)